crates/core/src/transport/connection_info.rs

```diff
@@ -0,0 +1,16 @@
+use aes_gcm::aes::Aes128;
+use std::net::SocketAddr;
+use thiserror::Error;
+
+use super::crypto::TransportPublicKey;
+
+pub(super) struct ConnectionInfo {
+    pub outbound_symmetric_key: Aes128,
```

Comment:
I notice that we're no-longer storing the `inbound_intro_packet` in the `ConnectionInfo`, the reason I was storing that is so that if a second intro packet is received it can be identified and acknowledged again, and then ignored - this might happen if the first acknowledgement is dropped. I was also storing the `outbound_intro_packet` so it could be resent without having to recreate it.

Did you have another idea in mind for handling this?

Reply:
Since the connection process is handled sequentially and at some point either you will get the intro packet and decrypt the key etc. or continue I don't see the need to keep that around for anything.

If an other intro packet comes out of order after the initial handshake has been done it can be just ignored, do we need it for anything?